scripts/EFI_workflow.R

```diff
@@ -0,0 +1,242 @@
+}
+
+# Do conversions
+#settings <- PEcAn.workflow::do_conversions(settings)
+
+# Write model specific configs
+if (PEcAn.utils::status.check("CONFIG") == 0) {
+  PEcAn.utils::status.start("CONFIG")
+  settings <-
+    PEcAn.workflow::runModule.run.write.configs(settings)
+  
+  PEcAn.settings::write.settings(settings, outputfile = "pecan.CONFIGS.xml")
+  PEcAn.utils::status.end()
+} else if (file.exists(file.path(settings$outdir, "pecan.CONFIGS.xml"))) {
+  settings <- PEcAn.settings::read.settings(file.path(settings$outdir, "pecan.CONFIGS.xml"))
+}
+
+if ((length(which(commandArgs() == "--advanced")) != 0)
+    && (PEcAn.utils::status.check("ADVANCED") == 0)) {
+  PEcAn.utils::status.start("ADVANCED")
+  q()
+}
+
+# Start ecosystem model runs
+if (PEcAn.utils::status.check("MODEL") == 0) {
+  PEcAn.utils::status.start("MODEL")
+  stop_on_error <- as.logical(settings[[c("run", "stop_on_error")]])
+  if (length(stop_on_error) == 0) {
+    # If we're doing an ensemble run, don't stop. If only a single run, we
+    # should be stopping.
+    if (is.null(settings[["ensemble"]]) ||
+        as.numeric(settings[[c("ensemble", "size")]]) == 1) {
+      stop_on_error <- TRUE
+    } else {
+      stop_on_error <- FALSE
+    }
+  }
+  PEcAn.remote::runModule.start.model.runs(settings, stop.on.error = stop_on_error)
+  PEcAn.utils::status.end()
+}
+
+# Get results of model runs
+if (PEcAn.utils::status.check("OUTPUT") == 0) {
+  PEcAn.utils::status.start("OUTPUT")
+  runModule.get.results(settings)
+  PEcAn.utils::status.end()
+}
+
+# Run ensemble analysis on model output.
+if ("ensemble" %in% names(settings)
+    && PEcAn.utils::status.check("ENSEMBLE") == 0) {
+  PEcAn.utils::status.start("ENSEMBLE")
+  runModule.run.ensemble.analysis(settings, TRUE)
+  PEcAn.utils::status.end()
+}
+
+# Run state data assimilation
+if ("state.data.assimilation" %in% names(settings)) {
+  if (PEcAn.utils::status.check("SDA") == 0) {
+    PEcAn.utils::status.start("SDA")
+    settings <- sda.enfk(settings)
+    PEcAn.utils::status.end()
+  }
+}
+
+# Pecan workflow complete
+if (PEcAn.utils::status.check("FINISHED") == 0) {
+  PEcAn.utils::status.start("FINISHED")
+  PEcAn.remote::kill.tunnel(settings)
+  db.query(
+    paste(
+      "UPDATE workflows SET finished_at=NOW() WHERE id=",
+      settings$workflow$id,
+      "AND finished_at IS NULL"
+    ),
+    params = settings$database$bety
+  )
+  
+  # Send email if configured
+  if (!is.null(settings$email)
+      && !is.null(settings$email$to)
+      && (settings$email$to != "")) {
+    sendmail(
+      settings$email$from,
+      settings$email$to,
+      paste0("Workflow has finished executing at ", base::date()),
+      paste0("You can find the results on ", settings$email$url)
+    )
+  }
+  PEcAn.utils::status.end()
+}
+
+db.print.connections()
+print("---------- PEcAn Workflow Complete ----------")
+
+
+#EFI Output Configuration
+library("ggplot2")
+library("plotly")
+library("gganimate")
+library("thematic")
```

Comment:
reference functions by namespace rather than using `library` loads

Reply:
Got it.